packages/api-derive/src/accounts/info.ts

```diff
@@ -7,39 +7,44 @@ import { AccountId, AccountIndex, Address, Balance } from '@polkadot/types/inter
 import { Codec } from '@polkadot/types/types';
 import { DeriveAccountInfo } from '../types';
 
-import { Observable, of } from 'rxjs';
+import { Observable, of, combineLatest } from 'rxjs';
 import { map, switchMap } from 'rxjs/operators';
 import { Bytes, Option, u32 } from '@polkadot/types';
 import { u8aToString } from '@polkadot/util';
 
-import { drr } from '../util/drr';
+import { drr, memo } from '../util';
 import { idAndIndex } from './idAndIndex';
 
+function retrieveNick (api: ApiInterfaceRx): (accountId?: AccountId) => Observable<Option<[Bytes, Balance] & Codec> | undefined> {
+  return (accountId?: AccountId): Observable<Option<[Bytes, Balance] & Codec> | undefined> =>
+    accountId && api.query.nicks
+      ? api.query.nicks.nameOf<Option<[Bytes, Balance] & Codec>>(accountId)
+      : of(undefined);
+}
+
 /**
  * @name info
  * @description Returns aux. info with regards to an account, current that includes the accountId, accountIndex and nickname
  */
 export function info (api: ApiInterfaceRx): (address?: AccountIndex | AccountId | Address | string | null) => Observable<DeriveAccountInfo> {
   const idAndIndexCall = idAndIndex(api);
+  const nickCall = memo(retrieveNick(api));
```

Comment:
Do you see a difference with memo like this?

It's unclear what memo does here: `retrieveNick(api)` creates a new function everytime, so everytime we cache a new function.

Intuitively i would do:
```
const retrieveNickMemo = memo(retrieveNick);
const nickCall = memo(retrieveNickMemo(api));
```

Reply:
Yes, that memo dropped the connections. I had a log inside the `retrieveNick`, initially it fired each time, then... well, it stopped. (It is most probably unneeded atm after the refCountDelay, have not checked, on 3 things at once)